File: py/338.py
# Author: Andrew Hamara

# Solution to leetcode problem 338. Counting bits

class Solution:
    def countBits(self, n:int) -> List[int]:
        bitCount = [0] * (n + 1)
        offset = 1
        for i in range(1, n + 1):
            if i == (offset * 2):
                offset = i
            bitCount[i] = 1 + bitCount[i - offset]
        return bitCount

File: go/338.go
func countBits(n int) []int {
    length := n + 1
    bits :=make([]int, length)
    offset := 1

    for i := 1; i < length; i++ {
        if i == offset * 2 {
            offset = i
        }
        bits[i] = 1 + bits[i-offset]
    }
    return bits
}
